src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -73,27 +73,31 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
     webapp_def = Site(location=location, site_config=site_config, server_farm_id=plan_info.id, tags=tags)
     helper = _StackRuntimeHelper(client, linux=is_linux)
 
-    if is_linux:
-        if not validate_linux_create_options(runtime, deployment_container_image_name,
-                                             multicontainer_config_type, multicontainer_config_file):
+    if is_linux or plan_info.is_xenon:
```

Comment:
i suggest you refactor, say fork out 2 dedicate methods one for linux and the other for windows, then write unit tests to make sure the configuration is right.
the code here is unreadable now with layers of patch